equinox/tree.py

```diff
@@ -304,3 +304,37 @@ def where(pytree):
         )
 
     return tree_at(_inferences, pytree, replace_fn=lambda _: value)
+
+
+def _ordered_tree_map(
+    f: Callable[..., Any],
+    tree: Any,
+    *rest: Any,
+    is_leaf: Optional[Callable[[Any], bool]] = None
+) -> Any:
+    """In an orderly fashion, maps a multi-input function over pytree args to produce a new pytree.
```

Comment:
I think this docstring is way too unwieldy. Something like "Like `jax.tree_util.tree_map`, but guaranteed to iterate over the tree in fixed order. (Namely depth-first left-to-right.)"